File: 02-Static-Array/LeetCode-1470.cpp
// 1470. Shuffle the Array
class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        for (int i = n; i < nums.size(); i ++)
            nums[i] = nums[i] * 1024 + nums[i-n];

        int idx = 0;
        for (int i = n; i < nums.size(); i ++)
        {
            nums[idx ++] = nums[i] % 1024;
            nums[idx ++] = nums[i] / 1024;
        }
        return nums;
    }
};

File: 02-Array/LeetCode-1470.py
# 1470. Shuffle the Array
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        for i in range(n, len(nums)):
            nums[i] = nums[i] * 1024 + nums[i-n]

        idx = 0
        for i in range(n, len(nums)):
            nums[idx] = nums[i] % 1024
            nums[idx + 1] = nums[i] // 1024
            idx += 2

        return nums    
